src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -105,6 +105,28 @@ object SnowflakeTelemetry {
     }
   }
 
+  private val MAX_CACHED_SPARK_PLAN_STATISTIC_COUNT = 1000
+
+  private[snowflake] def addSparkPlanStatistic(statistic: Set[String]): Unit = {
+    if (statistic.nonEmpty) {
+      this.synchronized {
+        if (logs.length >= MAX_CACHED_SPARK_PLAN_STATISTIC_COUNT) {
```

Comment:
If the pushdown is enabled, the spark plan statistic is added to `SnowflakeTelemetry.logs`, but it will be sent if any other kinds of messages is sent to snowflake lately.
In worst case, the pushdown is enabled, but the user doesn't not read/write from/to snowflake, all the spark plan statistic may be cached. To avoid OOM, SC cache at most 1000 messages.

Reply:
If the cache is full, can we flush it? Otherwise it seems we'll miss the statistic. I wonder if just returning that will cause us to miss a lot of stats.

Reply:
SC can't flush it, because addSparkPlanStatistic() is called in pushdown analysis, there is no connection or sfOptions yet. It's possible to miss some plans.
In the case spark connector is used, it may not be possible because pushdown analysis is before read/write.

Reply:
Can we try to flush it (with try/catch)? We should do our best to not miss messages. It seems like 1000 may not be a lot, since there can be many plans processed, and each plan can have a lot of nodes, and each node can have many expressions.

Another idea is to keep some counter of how many messages were missed/skipped and push that whenever we send the telemetry messages. That way we can at least have some idea if we are missing some messages and we can react in future releases.

Reply:
We have closed the connection once SC finished the query. I don't think we need to cache a connection for this purpose.
IMO, 1000 is a pretty big number because we only process Plan with `ReturnAnswer` on the top. No matter how complex a DataFrame is, only one message is generated. One `ReturnAnswer` is for one action.
I also think it is not necessary to track the missed_count, because it is low possibility that SC is used  after 1000+ non-spark-connector related actions.
The real world should be:
1. They don't use SC at all.
2. They use the SC a lot

Reply:
Ok, I see now that we add this only for `ReturnAnswer`. That should be ok.